### PR TITLE
631. Default roles are properly defined.

### PR DESCRIPTION
**Pull request by nekron**
Fixes 631

### CODE REVIEW

File: app/policies/roles/institution/health_organization.json.erb [hunk 1]
```diff
@@ -10,20 +10,14 @@
...
         {
-          "resource": [
-            "site?institution=<%= institution_id %>",
-            "device?institution=<%= institution_id %>",
-            "testResult?institution=<%= institution_id %>",
-            "encounter?institution=<%= institution_id %>"
-          ],
-          "action": "*",
+          "resource": "testResult?institution=<%= institution_id %>",
+          "action": "testResult:medicalDashboard",
```
Comment (spalladino) [inline app/policies/roles/institution/health_organization.json.erb]: The action checked for which test results are to be displayed in the dashboard is `testResult:query`; `testResult:medicalDashboard` is only checked to determine whether the user has access to the dashboard itself or not.

Comment (spalladino) [inline app/policies/roles/institution/health_organization.json.erb]: Also, note that NNDD currently does not support users having access to tests but not encounters. If we do have roles with this difference, then we'll need to push the permissions info to NNDD somehow.

File: app/policies/roles/institution/health_organization.json.erb [hunk 2]
```diff
@@ -10,20 +10,14 @@
             "institution:read",
             "institution:update",
             "institution:delete",
```
Comment (spalladino) [inline app/policies/roles/institution/health_organization.json.erb]: Is it ok to allow __any__ admin to delete the current institution?

File: app/policies/roles/institution/institution.json.erb [hunk 3]
```diff
@@ -42,9 +53,35 @@
...
+          "delegable": true
+        },
+        {
+          "resource": "device?institution=<%= institution_id %>",
+          "action": "device:read",
+          "delegable": true
+        },
+        {
+          "resource": "testResult?institution=<%= institution_id %>",
+          "action": "testResult:query",
```
Comment (spalladino) [inline app/policies/roles/institution/institution.json.erb]: No access to medical dashboard for institution readers?

File: app/policies/roles/institution/health_organization.json.erb [hunk 4]
```diff
@@ -40,11 +34,22 @@
...
           "action": [
-            "institution:read"
+            "institution:read",
+            "institution:readUsers"
           ],
           "delegable": true
+        },
+        {
+          "resource": "role?institution=<%= institution_id %>",
+          "action": "role:read",
```
Comment (spalladino) [inline app/policies/roles/institution/health_organization.json.erb]: What's the rationale for a reader to read roles?

File: app/policies/roles/institution/health_organization.json.erb [hunk 5]
```diff
@@ -40,11 +34,22 @@
     "policy": {
       "statement": [
         {
+          "resource": "testResult?institution=<%= institution_id %>",
+          "action": "testResult:medicalDashboard",
+          "delegable": true
```
Comment (spalladino) [inline app/policies/roles/institution/health_organization.json.erb]: Should all readers have delegable permissions?

File: app/policies/roles/site/institution.json.erb [hunk 6]
```diff
@@ -18,16 +39,37 @@
...
+          "delegable": true
+        },
+        {
+          "resource": "device?site=<%= site_id %>",
+          "action": "device:read",
+          "delegable": true
+        },
+        {
+          "resource": "testResult?site=<%= site_id %>",
+          "action": "testResult:query",
```
Comment (spalladino) [inline app/policies/roles/site/institution.json.erb]: Should an institution reader be able to access the medical dashboard?

File: app/policies/roles/site/institution.json.erb [hunk 7]
```diff
@@ -35,15 +77,38 @@
   },
   {
     "key": "technician",
-    "name": "Site <%=  %> Technician",
+    "name": "Site <%= site_name %> Technician",
```
Comment (spalladino) [inline app/policies/roles/site/institution.json.erb]: Is there an actual requirement for Technician and Supervisor?

File: app/policies/roles/institution/health_organization.json.erb (hunk 1 above)
Comment (nekron) [inline app/policies/roles/institution/health_organization.json.erb]: The point is that health organizations does not generate test results, so in order to view any in the dashboard they need to ask for that permission to another institution.

File: app/policies/roles/institution/health_organization.json.erb (hunk 2 above)
Comment (nekron) [inline app/policies/roles/institution/health_organization.json.erb]: Good point, maybe not.

File: app/policies/roles/site/institution.json.erb (hunk 7 above)
Comment (nekron) [inline app/policies/roles/site/institution.json.erb]: They were designed for Lidia and Raul, our technician and supervisor personas: https://docs.google.com/document/d/1BrDnV6FCDhZNaUTkhQXUL1U4nCQJ6D32lORRTnZJDxY/edit?pli=1#heading=h.lrp8tos9y27j
But there is no formal requirement of what they should be able to do if that's what you asked.

**Pull request merged**